src/components/PaymentLog.vue

```diff
@@ -580,9 +581,9 @@ export default {
 
   methods: {
     watchSalesLogForm(){
-      if(this.salesLogForm.sales_category_id == "2"){
+      if(this.salesLogForm.sales_category_id && this.salesLogForm.sales_category_id == "2"){
```

Comment:
why not try  this
`this.salesLogForm.discount = this.salesLogForm?.sales_category_id == "2" ? "5_discount" : "0_discount"`

Reply:
ok thanks